presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -185,13 +194,28 @@ private static boolean isColumnReference(QualifiedName name, RelationType relati
         return Optional.empty();
     }
 
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .addValue(relationId)
+                .toString();
+    }
+
     public static final class Builder
     {
+        private RelationId relationId = RelationId.anonymous();
```

Comment:
finally I think every scope should be attached to an AST node. If there are exceptions, those should be very, very rarely.